src/erlcloud_sns.erl

```diff
@@ -240,14 +242,14 @@ get_endpoint_attributes(EndpointArn, Config) ->
             Doc),
     [{arn, EndpointArn} | Decoded].
 
--spec get_endpoint_attributes/3 :: (string(), string(), string()) -> sns_endpoint().
+-spec get_endpoint_attributes/3 :: (parameter(), parameter(), parameter()) -> sns_endpoint().
 get_endpoint_attributes(EndpointArn, AccessKeyID, SecretAccessKey) ->
     get_endpoint_attributes(EndpointArn, new_config(AccessKeyID, SecretAccessKey)).
 
 
 
--spec set_endpoint_attributes/2 :: (string(), [{sns_endpoint_attribute(), string()}]) -> string().
--spec set_endpoint_attributes/3 :: (string(), [{sns_endpoint_attribute(), string()}], aws_config()) -> string().
+-spec set_endpoint_attributes/2 :: (parameter(), [{sns_endpoint_attribute(), parameter()}]) -> string().
+-spec set_endpoint_attributes/3 :: (parameter(), [{sns_endpoint_attribute(), parameter()}], aws_config()) -> string().
```

Comment:
Line 252 is too long: -spec set_endpoint_attributes/3 :: (parameter(), [{sns_endpoint_attribute(), parameter()}], aws_config()) -> string()..